src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs

```diff
@@ -261,11 +261,13 @@ public override void LoadXml(XmlElement element!!)
             XmlNamespaceManager nsm = new XmlNamespaceManager(element.OwnerDocument.NameTable);
             nsm.AddNamespace("ds", SignedXml.XmlDsigNamespaceUrl);
 
-            XmlNodeList x509IssuerSerialNodes = element.SelectNodes("ds:X509IssuerSerial", nsm);
-            XmlNodeList x509SKINodes = element.SelectNodes("ds:X509SKI", nsm);
-            XmlNodeList x509SubjectNameNodes = element.SelectNodes("ds:X509SubjectName", nsm);
-            XmlNodeList x509CertificateNodes = element.SelectNodes("ds:X509Certificate", nsm);
-            XmlNodeList x509CRLNodes = element.SelectNodes("ds:X509CRL", nsm);
+            //red flag - no existing null checks before usage, so should we assume that it's not null,
```

Comment:
Should we assume that `SelectNodes` will _not_ return null? Or should we check for null? Or should we do something else in this case?

Reply:
> Should we assume that `SelectNodes` will _not_ return null? Or should we check for null? Or should we do something else in this case?

Reply:
It's more important to get public APIs right than getting implementation details right. In this specific case select nodes can only return null if specific XmlNode you're running it on will return null when you create an XPathNavigator on it which I don't believe is a case here:
https://github.com/dotnet/runtime/blob/6a889d234267a4c96ed21d0e1660dce787d78a38/src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlNode.cs#L93-L97

Reply:
OK, thank you.